### PR TITLE
Fix duplicated rate limit in scheduler

### PR DESCRIPTION
**Pull request by resouer**
Ref #19012

Bind Pods rate limiter does not make too much sense when client already has limiter.

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot Test this please

**Comment by wojtek-t** [comment]
@hongchaodeng @xiang90 - FYI

**Comment by resouer** [comment]
@k8s-bot test this

**Comment by hongchaodeng** [comment]
I agree they are duplicate. But the default value of both are different.
How does it affect other dimensions of metrics (e.g. latency) with the change?

**Comment by wojtek-t** [comment]
The removed rate-limitting (BindPodQPS) didn't have any effect, because, we use 2 api calls per binding, so based on QPS limit we cannot bind more than 25 pods per second (50 in burst).
So this is "no-op" change from the whole system perspective.

**Comment by resouer** [comment]
And I'm afraid other components may also have duplication, gonna dig it thru after this.

**Comment by resouer** [comment]
@davidopp @wojtek-t Any review comments or label?

**Comment by wojtek-t** [comment]
Sorry - I'm on paternity leave. @gmarek - can you please take a look on Monday?

File: plugin/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -108,20 +104,11 @@ func New(c *Config) *Scheduler {
 
 // Run begins watching and scheduling. It starts a goroutine and returns immediately.
 func (s *Scheduler) Run() {
-	if s.config.BindPodsRateLimiter != nil {
-		go util.Forever(func() {
-			sat := s.config.BindPodsRateLimiter.Saturation()
-			metrics.BindingRateLimiterSaturation.Set(sat)
```
Comment (gmarek) [inline plugin/pkg/scheduler/scheduler.go]: Can you remove this metric?

**Comment by gmarek** [comment]
One thing. Generally LGTM.

**Comment by resouer** [comment]
@k8s-bot  test this

**Comment by resouer** [comment]
@gmarek Updated

**Comment by gmarek** [comment]
Thanks!

**Comment by gmarek** [comment]
@resouer please squash commits.

**Comment by gmarek** [comment]
Thanks

**Comment by gmarek** [comment]
@k8s-bot e2e test this

**Comment by resouer** [comment]
@k8s-bot unit test this please

**Comment by resouer** [comment]
@k8s-bot e2e test this

**Comment by resouer** [comment]
Not sure why e2e flaky here @k8s-bot e2e test this please

**Comment by mikedanese** [comment]
This is an API breaking change. We shouldn't remove command line flags without at least a release of deprecations. cc @davidopp @bgrant0607

**Comment by gmarek** [comment]
Yeah, you're right - I forgot that we should treat flags as API changes. I'll readd those flags to unbreak it.

**Pull request merged**